risc0/zkvm/src/serde/serializer.rs

```diff
@@ -33,6 +42,27 @@ pub trait WordWrite {
 }
 
 impl WordWrite for Vec<u32> {
+    #[inline]
+    fn start_new_buffered_word(&mut self) -> Result<()> {
+        self.push(0u32);
+        Ok(())
+    }
+
+    #[inline]
+    fn get_buffered_word(&self) -> Result<u32> {
+        let len = self.len();
+        assert_ne!(len, 0);
+        Ok(self[len - 1])
+    }
```

Comment:
there is a cost... which is the reason why I am thinking if I can avoid so. I agree it would be cleaner. How about using last but I unwrap immediately?

Reply:
Now obsolete in version 3.